src/lib.rs

```diff
@@ -147,6 +170,24 @@ pub enum SnmpError {
 
 pub type SnmpResult<T> = Result<T, SnmpError>;
 
+pub struct ResponseItem {
+    address: String, // like IPv4('192.168.1.1') or IPv6(_)
+    data: SnmpPdu,
+}
+
+impl fmt::Debug for ResponseItem {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let mut res = write!(f, "IP({}):\r\n", self.address.to_string());
```

Comment:
`\r` should not be used at all and `\n` is usually only used in complex structures. In this particular case I suggest to have `address` be of a `SocketAddr` and debug-format it with `"{:?}".

Reply:
I will remove '\r' and want to save the 'n' for good readability. It is a complex structure, because we have one IP and many oid-Values from it.

Reply:
By convention simple debug output is formatted as one-liner, if `Formatter#alternate` flag is set (e.g. {:#?}) then it can be multiline.